src/helpers/warning.ts

```diff
@@ -0,0 +1,10 @@
+import w from 'warning';
```

Comment:
I know this is maybe out of scope for this PR but given that we completely scope warnings not to be in production, we can probably just log ourselves here and get rid of the dependency

Reply:
I don't mind replacing the `w` with `console.error` or `console.warn` if you want. The current behaviour of `warning` is to call error